src/zjs_fs.c

```diff
@@ -510,6 +522,14 @@ static jerry_value_t zjs_write(const jerry_value_t function_obj,
     if (!jerry_get_object_native_handle(argv[0], (uintptr_t*)&handle)) {
         return zjs_error("native handle not found");
     }
+
+    if (handle->mode == MODE_R) {
+        return zjs_error("file is not open for writing");
+    }
+    if (handle->mode == MODE_A && position) {
+        return zjs_error("cannot seek in mode a");
```

Comment:
Actually the nodejs docs for write() explicitly state this is ignored, not an error, for the append case.

So actually you don't need to do this check at all here.